### ISSUE TITLE
Service Identity private_key is null

### ISSUE THREAD

**Issue by gertjanmaas** [philips-software/terraform-provider-hsdp#72]
A module creates a Service Identity and it's `private_key` is outputted from the module I created.

This morning after I had some issues with the service identity not being in the correct group (and couldn't manually add it via the self-service portal either), I decided to destroy my environment and rebuild it, to check if it's a fluke.

The destroy went fine, but when I wanted to rebuild my environment I received the following error:
```
Error: Invalid function argument
  on ../../layers/hsdp/org/outputs.tf line 10, in output "iam_service_private_key":
  10:   value = replace(replace(hsdp_iam_service.iam_service.private_key, "-----END", "\n-----END"), "KEY-----", "KEY-----\n")
    |----------------
    | hsdp_iam_service.iam_service.private_key is null
Invalid value for "str" parameter: argument must not be null.
```

After downgrading from 0.14.7 to 0.14.6 I could apply the environment again.

**Comment by loafoe** [comment]
In `0.14.6` the attribute was "calculated" whereis in .7 it is optional. I'll try to reproduce it here.

**Comment by loafoe** [comment]
I've reproduced the output you are seeing, however after reading up a bit on the Terraform core flow, it seems `output` values are read/refreshed before the apply step for some reasons. There's quite a few issues/discussions on the subject, but the issue/comment that describes the observed behaviour can be seen here:

https://github.com/hashicorp/terraform/issues/23365#issuecomment-554160790

A workaround for this:

```hcl
output "private_key" {
  value = replace(replace(hsdp_iam_service.test_service.private_key != null ? 
     hsdp_iam_service.test_service.private_key : "", "-----END", "\n-----END"), "KEY-----", "KEY-----\n")
  sensitive = true
}
```

Alternatively, you could "fix" the `private_key` in the consuming module.

**Comment by loafoe** [comment]
@gertjanmaas release `0.14.8` pre-formats the generated IAM key before storing it in Terraform state. This means you no longer have to perform the replace and the key can be fed to your parser as is.

**Comment by gertjanmaas** [comment]
Fixed in `0.14.8`
